apps/meteor/client/views/room/hooks/useOpenRoom.ts

```diff
@@ -88,18 +91,35 @@ export function useOpenRoom({ type, reference }: { type: RoomType; reference: st
 			unsubscribeFromRoomOpenedEvent.current();
 			unsubscribeFromRoomOpenedEvent.current = RoomManager.once('opened', () => fireGlobalEvent('room-opened', omit(room, 'usernames')));
 
+			const sub = Subscriptions.findOne({ rid: room._id });
+			const isAnonymous = !user;
+
+			console.log(isAnonymous);
+			console.log(sub);
+			console.log(hasPreviewPermission);
+			console.log(hasPermission('preview-c-room'));
+
+			if (isAnonymous && !hasPreviewPermission) {
```

Comment:
Anonymous users should be able to subscribe regardless of the permission.

Change this to not check the preview permission if both:
1. there is no userId
2. anonymous-read is enabled.

anonymous permissions are a mess and we should probably improve it on the backend to validate the permissions of the anonymous role automatically, but for now to keep it consistent with everything else, we need to skip some client side permission checks for anonymous users.